htdocs/comm/propal/class/propalestats.class.php

```diff
@@ -248,7 +248,7 @@ public function getAllByProduct($year, $limit = 10)
 		global $user;
 
 		$sql = "SELECT product.ref, COUNT(product.ref) as nb, SUM(tl.".$this->field_line.") as total, AVG(tl.".$this->field_line.") as avg";
-		$sql .= " FROM ".$this->from.", ".$this->from_line.", ".MAIN_DB_PREFIX."product as product";
+		$sql .= " FROM (".$this->from.", ".$this->from_line.", ".MAIN_DB_PREFIX."product as product)";
```

Comment:
The syntax using the ( ) does not works on all database version.
The good solution is to have all INNER and OUTER JOIN always just after the table so
```
$sql .= " FROM ".$this->from;
if (empty($user->rights->societe->client->voir) && !$user->socid) {
			$sql .= "  INNER JOIN ".MAIN_DB_PREFIX."societe_commerciaux as sc ON p.fk_soc = sc.fk_soc AND sc.fk_user = ".((int) $user->id);
		}
$sql .=", ".$this->from_line.", ".MAIN_DB_PREFIX."product as product";
```

Reply:
I replaced my earlier version using parentheses with a new proposal using only explicit inner joins: would this be ok?

Reply:
Yes, using inner join is also possible.